packages/react-router/lib/hooks.tsx

```diff
@@ -544,10 +545,12 @@ export class RenderErrorBoundary extends React.Component<
 
   render() {
     return this.state.error ? (
-      <RouteErrorContext.Provider
-        value={this.state.error}
-        children={this.props.component}
-      />
+      <RouteContext.Provider value={this.props.routeContext}>
+        <RouteErrorContext.Provider
+          value={this.state.error}
+          children={this.props.component}
+        />
+      </RouteContext.Provider>
```

Comment:
Need to put the error boundary inside the `RouteContext` so it reflects the right matches, otherwise it thinks it' at the parent layer in the nested hierarchy 😬